internal/core/docker_host.go

```diff
@@ -56,6 +56,23 @@ func DefaultGatewayIP() (string, error) {
 	return ip, nil
 }
 
+// Use a vanilla Docker client to check if the Docker host is reachable.
+// It will avoid recursive calls to this function.
+var defaultCallbackCheckFn = func(host string) error {
```

Comment:
suggestion: I don't think we need this to be a var.
```suggestion
func defaultCallbackCheck(host string) error {
```

Reply:
Ignore this if the structure change above makes sense, but would rename as per example to something like `hostCheck`.